src/main.rs

```diff
@@ -313,6 +313,11 @@ fn find_ram_in_linker_script(linker_script: &str) -> Option<MemoryEntry> {
     for (index, mut line) in linker_script.lines().enumerate() {
         line = line.trim();
         line = eat!(line, "RAM");
+
+        while !line.starts_with(":") {
+            line = line[1..].trim()
```

Comment:
indexing a `str` like can panic because the index is in bytes, and not in UTF8 code points (`str[1..]` will panic if the first character is not an ascii character)
could you change this to skip directly to the `:` character?
I think you can use the `str::char_indices` method + `Iterator::find` to find the byte location of `:` then you can use that location to slice the string. That way we are sure the index will fall on a `char` boundary

Reply:
used str::find()